lib/taskana-core/src/main/java/pro/taskana/classification/internal/util/ClassificationQueryMapperUtil.java

```diff
@@ -0,0 +1,166 @@
+package pro.taskana.classification.internal.util;
+
+public class ClassificationQueryMapperUtil {
+  private static final String OPENING_SCRIPT_TAG = "<script>";
+  private static final String CLOSING_SCRIPT_TAG = "</script>";
+  private static final String OPENING_WHERE_TAG = "<where>";
+  private static final String CLOSING_WHERE_TAG = "</where>";
+
+  private static final String SELECT_COUNT_ID = "SELECT COUNT(ID)";
+  private static final String SELECT_DISTINCT_COLUMN_NAME = "SELECT DISTINCT ${columnName} ";
+  private static final String FROM_CLASSIFICATION = " FROM CLASSIFICATION ";
+
+  private static final String DB2_CONDITION = "<if test=\"_databaseId == 'db2'\">with UR </if> ";
+
+  public String getClassificationSummariesQuery() {
+    return OPENING_SCRIPT_TAG
+        + getSelectColumns()
+        + FROM_CLASSIFICATION
+        + getWhereConditions()
+        + getOrderBy()
+        + DB2_CONDITION
+        + CLOSING_SCRIPT_TAG;
+  }
+
+  public String getCountClassificationsQuery() {
+    return OPENING_SCRIPT_TAG
+        + SELECT_COUNT_ID
+        + FROM_CLASSIFICATION
+        + getWhereConditions()
+        + DB2_CONDITION
+        + CLOSING_SCRIPT_TAG;
+  }
+
+  public String getClassificationColumnValuesQuery() {
+    return OPENING_SCRIPT_TAG
+        + SELECT_DISTINCT_COLUMN_NAME
+        + FROM_CLASSIFICATION
+        + getWhereConditions()
+        + getOrderBy()
+        + DB2_CONDITION
+        + CLOSING_SCRIPT_TAG;
+  }
+
+  private String getSelectColumns() {
+    return "SELECT ID, KEY, PARENT_ID, PARENT_KEY, CATEGORY, TYPE, DOMAIN, VALID_IN_DOMAIN, "
+        + "CREATED, NAME, DESCRIPTION, PRIORITY, SERVICE_LEVEL, APPLICATION_ENTRY_POINT, "
+        + "CUSTOM_1, CUSTOM_2, CUSTOM_3, CUSTOM_4, CUSTOM_5, CUSTOM_6, CUSTOM_7, CUSTOM_8";
+  }
+
+  private String getWhereConditions() {
+    return OPENING_WHERE_TAG
+        + getClassificationQueryInConditions()
+        + getClassificationQueryLikeConditions()
+        + CLOSING_WHERE_TAG;
+  }
+
+  private String getOrderBy() {
+    return "<if test='!orderBy.isEmpty()'>ORDER BY "
+        + "<foreach item='item' collection='orderBy' separator=',' >${item}</foreach></if> ";
+  }
+
+  private StringBuilder getClassificationQueryInConditions() {
+    final StringBuilder query = new StringBuilder(255);
```

Comment:
BTW: why do we use String concatination at one place and StringBuffer at another place?

Reply:
It was a one-liner, which was split in 2 lines on "format".
I didn't want to write 
``return new StringBuilder("");``
for just one line.